src/mysql/MySQL.chpl

```diff
@@ -0,0 +1,611 @@
+        this._nFields = mysql_num_fields(this._cptr_result): int(32);
+        this._cptr_fields = mysql_fetch_fields(this._cptr_result);
+        
+        this._nRows = mysql_num_rows(this._cptr_result);
+      }
+
+      else {
+        writeln("[Error] Error occurred while storing result.");
+        throw new QueryExecutionError();
+        writeln(createStringWithNewBuffer(mysql_error(this._cptr_mysqlconn)));
+      }
+    }
+
+    /*
+    Execute a batch of SQL statements in a single transaction.
+    Commits after their execution, or rolls back if there was an error.
+      :arg statements: an array of :class:`Statement` objects to be executed
+      :type statements: array of :class:`Statement` objects
+    */
+    override proc executeBatch(statements: [] owned Statement) throws {
+      this._connection.beginTransaction();
+      var hasError: bool = false;
+
+      // TODO: memory management
+      for stmt in statements {
+        try {
+          this.execute(stmt);
+        }
+        catch err: QueryExecutionError {
+          hasError = true;
+          this._connection.rollback();
+
+          // re-throw the query execution error
+          throw new QueryExecutionError();
+        }
+      }
+
+      if (!hasError) {
+        this._connection.commit();
+      }
+    }
+
+    /*
+    Fetches the next row of the result set.
+    Returns a row whcih gives false on calling isValid() on it.
+    */
+    override proc fetchone(): owned MySQLRow? {
+      if (this._curRow >= this._nRows) {
+        return nil;
+      }
+
+      var nextRow: MYSQL_ROW = mysql_fetch_row(this._cptr_result);
+      // TODO: is there any way to check if nextRow is NULL here?
+      // using nextRow == nil or nextRow == c_nil don't work
+
+      this._curRow += 1;
+
+      // init the row:
+      var _row = new MySQLRow(nextRow, this._cptr_fields);
+      
+      return _row;
+    }
+
+    /*
+    Fetches the next given number of rows in the result one by one,
+    returning an iterator.
+    Throws Error if howManyRows is negative.
+    In case howManyRows > actual number of rows in the retrieved set, 
+    all the rows are returned.
+
+      :args howManyRows: how many rows to return
+      :type howManyRows: int(32)
+    */
+    override iter fetchsome(howManyRows: int(32)): owned MySQLRow? {
+      if (howManyRows < 0) {
+        writeln("[Error] MySQLCursor.fetchsome(howManyRows) called with howManyRows < 0");
+        throw new Error();
+      }
+
+      if (howManyRows > this._nRows) {
+        writeln("[Warning] MySQLCursor.fetchsome(howManyRows) called with howManyRows > _nRows; whole result set will be returned.");
+      }
+
+      else {
+        var counter = 0;
+        var _row = this.fetchone();
+
+        while (_row != nil && counter < this._nRows && counter < howManyRows) {
+          yield _row;
+          _row = this.fetchone();
+          counter += 1;
+        }
+      }
+    }
+
```

Comment:
Doesn't look like you are returning any row in case of howManyRows > actual number of rows.

Reply:
Will fix this